### DISCUSSION TITLE
Security - Method setAuthenticated deprecated, do i still have to use it until Symfony 6+ ?

### DISCUSSION THREAD

**Discussion by YannickDurden** [symfony/symfony#50603] [Q&A]
Hi,

Following the snippet below:

```php
$userBadge = new UserBadge($tokens['token'], function () use ($authenticatedToken) {
    $user = $this->userManager->me();
    $authenticatedToken->setUser($user);
    $authenticatedToken->setAuthenticated(true);
    $this->tokenStorage->setToken($authenticatedToken);
    return $user;
});
```
How I can replace the setAuthenticated method? Because if I stop to using it, my user is not logged.

The variable `$authenticatedToken` is an instance of JwtToken, which one extends AbstractToken. And in this class, we can find `$authenticated` with some methods.

Or maybe, if I'm using the deprecated method, when I'll update to Symfony 6+, the login still works?

**Comment by MatTheCat** [comment]
Hi,

you have to call `TokenInterface::setAuthenticated` as long as it exists. Since it is no longer the case with Symfony 6, removing the call should be enough when you’ll upgrade: https://github.com/symfony/symfony/blob/a5e6d8d1d330cb1d5ab6b4bc76dc9439b4b4e302/src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php#L74

**Reply by YannickDurden** [reply]
Simple! Thanks for the reply 👍

**Comment by ghost** [comment]
`setAuthenticated` is removed in SF 6. Return `null` from `getUser()` instead when a token is not authenticated

**Reply by YannickDurden** [reply]
Yes, I read it in the above answer. Look's fine for me
